src/services/ApiService.tsx

```diff
@@ -0,0 +1,100 @@
+import { Observable } from "rxjs";
+import Weather from "../models/Weather";
+import ApiInfo from './ApiInfo';
+import HourlyForecast from "../models/HourlyForecast";
+import { HttpService } from "./HttpService";
+/*
+  The purpose of this class is to be the main point of API consume.
+  This class will be used as the Service that is allowed to request using REST to OpenWeather API
+ */
+export class ApiService {
+    private provider: HttpService;
+
+    constructor() { 
+      this.provider = new HttpService();
+    }
+  
+    /**
+     * This method returns the weather for the given city
+     * @param cityName Name of the city
+     */
+    getCurrentWeather(cityName: string): Observable<Weather> {
+      return Observable.create((observer: { next: (arg0: any) => void; }) => {
```

Comment:
There are lots of `arg0` throughout the code. Could you rename them to a more meaningful name?

Reply:
Any plans to refactor these argument names?